src/compiler/crystal/semantic/top_level_visitor.cr

```diff
@@ -1054,7 +1054,11 @@ class Crystal::TopLevelVisitor < Crystal::SemanticVisitor
 
   def check_ditto(node)
     stripped_doc = node.doc.try &.strip
-    if stripped_doc == ":ditto:" || stripped_doc == "ditto"
+    if stripped_doc == ":ditto:"
+      node.doc = @last_doc
```

Comment:
Both branches assign `node.doc = @last_doc` and afterwards there's `@last_doc = node.doc`. This looks like it could be simplified. The last assignment only has an effect when neither condition matches.

And, more generally we don't even care about docs when `@program.wants_doc?` is `false`. So that could be an early return for this method.